internal/controller/apiextensions/composite/composed_test.go

```diff
@@ -234,6 +235,35 @@ func TestRender(t *testing.T) {
 				err: errors.Wrap(errBoom, errName),
 			},
 		},
+		"ControllerError": {
+			reason: "External controller owner references should cause an exception",
+			client: &test.MockClient{MockCreate: test.NewMockCreateFn(nil)},
+			args: args{
+				cp: &fake.Composite{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{
+					xcrd.LabelKeyNamePrefixForComposed: "ola",
+					xcrd.LabelKeyClaimName:             "rola",
+					xcrd.LabelKeyClaimNamespace:        "rolans",
+				}}},
+				cd: &fake.Composed{ObjectMeta: metav1.ObjectMeta{Name: "cd",
+					OwnerReferences: []metav1.OwnerReference{{Controller: &ctrl,
+						UID: "random_uid"}}}},
+				t: v1.ComposedTemplate{Base: runtime.RawExtension{Raw: tmpl}},
+			},
+			want: want{
+				cd: &fake.Composed{ObjectMeta: metav1.ObjectMeta{
+					Name:         "cd",
+					GenerateName: "ola-",
+					Labels: map[string]string{
+						xcrd.LabelKeyNamePrefixForComposed: "ola",
+						xcrd.LabelKeyClaimName:             "rola",
+						xcrd.LabelKeyClaimNamespace:        "rolans",
+					},
+					OwnerReferences: []metav1.OwnerReference{{Controller: &ctrl,
+						UID: "random_uid"}},
+				}},
+				err: errors.Wrap(errors.Errorf("cd is already controlled by   (UID random_uid)"), errSetControllerRef),
```

Comment:
The format of this error returned by `meta.AddControllerReference` seems like it might be a bug. 🤔

Reply:
Would you like me to reformat?

Reply:
No - just thinking out loud. I'm pretty sure this would need to be fixed in crossplane-runtime. You're welcome to make that fix if you like but it definitely doesn't need to block this PR.